File: day01/part1.py
input = open('./input.txt', 'r')
data = input.readlines()

frequency = 0
for frequency_change in data:
    try:
        change = int(frequency_change.rstrip())
        if change:
            frequency += change
    except:
        print('Bad value: {}'.format(repr(frequency_change)))

print('frequency: {}'.format(frequency))

File: day01/elixir/part1.exs
input = 
  File.stream!('../input.txt')
  |> Stream.map(&String.trim_trailing(&1) |> String.to_integer)
  |> Enum.to_list

frequency = Enum.reduce(input, 0, fn(frequency_change, acc) -> acc + frequency_change end)

IO.puts 'frequency: #{frequency}'
